src/pages/login/login.js

```diff
@@ -2,37 +2,38 @@ import React, { useState } from 'react';
 import './login.css';
 import 'styles.css';
 import { Link, useNavigate } from 'react-router-dom';
-import { useAuth } from 'context/authContext';
+import { userLogin } from '../../redux/slices/userSlice';
+import { useDispatch, useSelector } from 'react-redux';
 
 export default function Login() {
     const [email, setEmail] = useState();
     const [password, setPassword] = useState();
-    const { LoginUser } = useAuth();
     const navigate = useNavigate();
-    const handleLogin = async (e) => {
+    const dispatch = useDispatch();
+
+    const handleSubmit = async (e) => {
         e.preventDefault();
         if (email === undefined || password === undefined) {
             alert("Please enter email and password.");
         }
         else {
-            const LoginResponse = await LoginUser(email, password);
-            if (LoginResponse) {
+
+            const res = await dispatch(userLogin({ email: email, password: password }))
+            if (res.payload.uid) {
                 navigate("/explore");
             } else {
-                alert("Invalid credentials, please sign up.");
+                alert(res.payload);
```

Comment:
Good logic of raising errors from the dispatch response!! Awesome work @shivanigangadharan